iree/compiler/Codegen/LLVMGPU/LLVMGPUUtils.h

```diff
@@ -30,6 +32,32 @@ static llvm::SmallVector<mlir::linalg::ProcInfo, 2> getGPUThreadIdsAndCounts(
   return procInfo;
 }
 
+/// Compute subgroup ID. CUDA doesn't have a subgroupId equivalent so we are are
```

Comment:
The impl should be in a .cpp file?

Reply:
So this has a strong assumption that each warp is responsible for `(X, Y, Z)` = `(<warp-size>, <full-Y>, <full-Z>)` threads. It would be great to specify that explicitly in the comment to make it clear for the curious. :)

Reply:
> The impl should be in a .cpp file?

True, I currently don't have a cpp file. I'd rather send a RFC change follow this PR as this PR is already big and I would rather not include extra refactoring changes. Is that okay with you?

> So this has a strong assumption that each warp is responsible for `(X, Y, Z)` = `(<warp-size>, <full-Y>, <full-Z>)` threads. It would be great to specify that explicitly in the comment to make it clear for the curious. :)

Good point added a comment.